File: tools/source/sdks/lua-sdk/examples/continuity_inspect.lua
local Client = require('client')

local client = Client.new(os.getenv('GAE_SERVER') or 'http://127.0.0.1:8080', os.getenv('GAE_KEY') or 'dev-key')
local world_id = os.getenv('GAE_WORLD_ID') or 'demo_world'
local log_limit = tonumber(os.getenv('GAE_LOG_LIMIT') or '10')
local trace_limit = tonumber(os.getenv('GAE_TRACE_LIMIT') or '10')
local timeline_limit = tonumber(os.getenv('GAE_TIMELINE_LIMIT') or '5')

local function dump(label, req)
  print('== ' .. label .. ' ==')
  print(req.method .. ' ' .. req.path)
  if req.body then
    print(req.body)
  end
end

dump('world settings', client:world_settings_request(world_id))
dump('state components', client:state_components_request(world_id))
dump('latest timeline', client:latest_timeline_request(world_id))
dump('recent timelines', client:timelines_request(world_id, timeline_limit))
dump('recent logs', client:logs_request(world_id, log_limit, 0, nil))
dump('recent debug traces', client:debug_traces_request(world_id, trace_limit))
dump('world policy', client:world_policy_request(world_id))

File: tools/source/sdks/c-sdk/examples/continuity_inspect.c
#include <stdio.h>
#include "../src/game_agent_engine_client.h"

int main(void) {
    const char* world_id = "demo_world";
    printf("== world settings ==\nGET %s\n", gae_world_settings_path(world_id));
    printf("== state components ==\nGET %s\n", gae_state_components_path(world_id));
    printf("== latest timeline ==\nGET %s\n", gae_latest_timeline_path(world_id));
    printf("== recent timelines ==\nGET %s\n", gae_timelines_path(world_id, 5));
    printf("== recent logs ==\nGET %s\n", gae_logs_path(world_id, 10, 0, NULL));
    printf("== recent debug traces ==\nGET %s\n", gae_debug_traces_path(world_id, 10));
    printf("== world policy ==\nGET %s\n", gae_world_policy_path(world_id));
    return 0;
}
